File: app/src/main/java/com/kimjisub/launchpad/midi/driver/MasterKeyboard.java
package com.kimjisub.launchpad.midi.driver;

public class MasterKeyboard extends DriverRef {
	@Override
	public void getSignal(int cmd, int sig, int note, int velo) {

		int x;
		int y;

		if (cmd == 9) {
			if (note >= 36 && note <= 67) {
				x = (67 - note) / 4 + 1;
				y = 4 - (67 - note) % 4;
				onPadTouch(x - 1, y - 1, velo != 0, velo);
			} else if (note >= 68 && note <= 99) {
				x = (99 - note) / 4 + 1;
				y = 8 - (99 - note) % 4;
				onPadTouch(x - 1, y - 1, velo != 0, velo);
			}

		} else if (velo == 0) {
			if (note >= 36 && note <= 67) {
				x = (67 - note) / 4 + 1;
				y = 4 - (67 - note) % 4;
				onPadTouch(x - 1, y - 1, false, 0);
			} else if (note >= 68 && note <= 99) {
				x = (99 - note) / 4 + 1;
				y = 8 - (99 - note) % 4;
				onPadTouch(x - 1, y - 1, false, 0);
			}
		}
	}

	@Override
	public void sendPadLED(int x, int y, int velo) {
	}

	@Override
	public void sendChainLED(int c, int velo) {
	}

	@Override
	public void sendFunctionkeyLED(int f, int velo) {
	}

	@Override
	public void sendClearLED() {
	}

}

File: app/src/main/java/com/kimjisub/launchpad/midi/driver/MasterKeyboard.kt
package com.kimjisub.launchpad.midi.driver

class MasterKeyboard : DriverRef() {
	companion object;

	override fun getSignal(cmd: Int, sig: Int, note: Int, velocity: Int) {
		val x: Int
		val y: Int
		if (cmd == 9) {
			if (note in 36..67) {
				x = (67 - note) / 4 + 1
				y = 4 - (67 - note) % 4
				onPadTouch(x - 1, y - 1, velocity != 0, velocity)
			} else if (note in 68..99) {
				x = (99 - note) / 4 + 1
				y = 8 - (99 - note) % 4
				onPadTouch(x - 1, y - 1, velocity != 0, velocity)
			}
		} else if (velocity == 0) {
			if (note in 36..67) {
				x = (67 - note) / 4 + 1
				y = 4 - (67 - note) % 4
				onPadTouch(x - 1, y - 1, false, 0)
			} else if (note in 68..99) {
				x = (99 - note) / 4 + 1
				y = 8 - (99 - note) % 4
				onPadTouch(x - 1, y - 1, false, 0)
			}
		}
	}

	override fun sendPadLed(x: Int, y: Int, velocity: Int) {}
	override fun sendChainLed(c: Int, velocity: Int) {}
	override fun sendFunctionkeyLed(f: Int, velocity: Int) {}
	override fun sendClearLed() {}
}